---
string[] varNames
string[] varTypes
int32 size
